movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+        all individuals between the given pair of keypoints, or if
+        multiple pairs are provided, a dictionary containing the computed
+        distances for each pair of keypoints, with the key being the pair
+        of keypoints and the value being the :class:`xarray.DataArray`
+        containing the computed distances.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) for all keypoints
+    between ``ind1`` and ``ind2`` (i.e. interindividual distance):
+
+    >>> position = xr.DataArray(
+    ...     np.arange(24).reshape(2, 2, 3, 2),
+    ...     coords={
+    ...         "time": np.arange(2),
+    ...         "individuals": ["ind1", "ind2"],
+    ...         "keypoints": ["key1", "key2", "key3"],
+    ...         "space": ["x", "y"],
+    ...     },
+    ...     dims=["time", "individuals", "keypoints", "space"],
+    ... )
+    >>> dist_key1_key2 = compute_interkeypoint_distances(
+    ...     position, pairs={"key1": "key2"}
+    ... )
+    >>> dist_key1_key2
+    <xarray.DataArray (time: 2, key1: 2, key2: 2)> Size: 64B
+    2.828 11.31 5.657 2.828 2.828 11.31 5.657 2.828
+    Coordinates:
+      * time     (time) int32 8B 0 1
+      * key1     (key1) <U4 32B 'ind1' 'ind2'
+      * key2     (key2) <U4 32B 'ind1' 'ind2'
+
+    The resulting ``dist_key1_key2`` is a DataArray containing the computed
+    distances between ``key1`` and ``key2`` for all individuals
+    at each time point.
+
+    To obtain the distances between ``key1`` and ``key2`` within ``ind1``:
+
+    >>> dist_key1_key2.sel(key1="ind1", key2="ind1")
+
+    To obtain the distances between ``key1`` of ``ind1`` and
+    ``key2`` of ``ind2``:
+
+    >>> dist_key1_key2.sel(key1="ind1", key2="ind2")
+
+    Compute the city block or Manhattan distance for multiple pairs of
+    keypoints using ``position``:
+
+    >>> key_dists = compute_interkeypoint_distances(
+    ...     position,
+    ...     pairs={"key1": "key2", "key3": ["key1", "key2"]},
+    ...     metric="cityblock",
+    ... )
+    >>> key_dists.keys()
+    dict_keys(['dist_key1_key2', 'dist_key3_key1', 'dist_key3_key2'])
+
+    As multiple pairs of keypoints are specified,
+    the resulting ``key_dists`` is a dictionary containing the DataArrays
+    of computed distances for each pair of keypoints.
+
+    Compute the city block or Manhattan distance for all possible pairs of
+    keypoints using ``position``:
+
+    >>> key_dists = compute_interkeypoint_distances(
+    ...     position,
+    ...     metric="cityblock",
+    ... )
+    >>> key_dists.keys()
+    dict_keys(['dist_key1_key2', 'dist_key1_key3', 'dist_key2_key3'])
+
+    See Also
+    --------
+    compute_interindividual_distances : Compute distances between one or
+        more pairs of individuals within and across all keypoints.
+
+    """
+    return _compute_pairwise_distances(
+        data, "keypoints", pairs=pairs, metric=metric
+    )
+
+
+def _compute_pairwise_distances(
```

Comment:
I think this function should be made public, and we should get rid of `compute_interindividual_distances` and `compute_interkeypoint_distances`. Their examples can be transferred to this docstring.

I will elaborate on the rationale in my review summary.